### PR TITLE
Improve Debug and Display for PeerId

### PR DESCRIPTION
**Pull request by tomaka**

### CODE REVIEW

File: core/src/peer_id.rs [hunk 1]
```diff
@@ -35,7 +35,15 @@ pub struct PeerId {
...
-        write!(f, "PeerId({})", self.to_base58())
+        f.debug_tuple("PeerId")
+            .field(&self.to_base58())
+            .finish()
+    }
+}
+
+impl fmt::Display for PeerId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Display::fmt(&self.to_base58(), f)
```
Comment (twittner) [inline core/src/peer_id.rs]:
```suggestion
        self.to_base58().fmt(f)
```

Comment (tomaka) [reply core/src/peer_id.rs]: Why is that better?

Comment (twittner) [reply core/src/peer_id.rs]: It looks less noisy to me and is shorter.

Comment (tomaka) [reply core/src/peer_id.rs]: But if we imported `Debug` and `Display` directly instead of through `fmt`, we'd get a compilation error.

Comment (twittner) [reply core/src/peer_id.rs]: Sure, but we did not.

**Pull request merged**